### PR TITLE
[WIP] Fix Apache wicket app analysis failure

### PR DESCRIPTION
**Pull request by nachandr**

### CODE REVIEW

**Comment by nachandr** [comment]
Apache_wicket

**Review by mguetta1 — CHANGES_REQUESTED** [review]
Thanks @nachandr for the PR.
Could you please add analysis tag and dependencies checks?

File: analysis/tc_apache_wicket.go [hunk 1]
```diff
@@ -22,16 +22,46 @@ var ApacheWicket = TC{
 		"cloud-readiness",
```
Comment (mguetta1) [inline analysis/tc_apache_wicket.go:22]: Lines 16-22 can be removed. If I am not mistaken those are leftovers from windup analysis

**Review by mguetta1 — CHANGES_REQUESTED** [review]
@nachandr please create a new application var for this analysis in https://github.com/konveyor/go-konveyor-tests/blob/main/data/application.go

**Comment by nachandr** [comment]
> Thanks @nachandr for the PR. Could you please add analysis tag and dependencies checks?

Analysis tags and dependencies are not generated with the selected target.

**Comment by mguetta1** [comment]
> > Thanks @nachandr for the PR. Could you please add analysis tag and dependencies checks?
> 
> Analysis tags and dependencies are not generated with the selected target.

Good, please set the expected result to an empty array so if something changes we can catch it - `Dependencies: []api.TechDependency{}`
Thanks!

File: analysis/tc_apache_wicket.go (hunk 1 above)
Comment (nachandr) [reply analysis/tc_apache_wicket.go]: @mguetta1 , done "(y)"

**Comment by mguetta1** [comment]
> > > Thanks @nachandr for the PR. Could you please add analysis tag and dependencies checks?
> > 
> > 
> > Analysis tags and dependencies are not generated with the selected target.
> 
> Good, please set the expected result to an empty array so if something changes we can catch it - `Dependencies: []api.TechDependency{}` Thanks!
> 
> Thanks @mguetta1. Just want to make sure I have this right. Are you referring to a placeholder for the future or something else? If it's just a placeholder , I think we could leave the code as is and add the dependencies when they do get generated.

Hi, not a placeholder but an empty array to make sure that this is the expected behavior instead of skipping the `Dependencies` check

**Comment by nachandr** [comment]
> > > > Thanks @nachandr for the PR. Could you please add analysis tag and dependencies checks?
> > > 
> > > 
> > > Analysis tags and dependencies are not generated with the selected target.
> > 
> > 
> > Good, please set the expected result to an empty array so if something changes we can catch it - `Dependencies: []api.TechDependency{}` Thanks!
> > Thanks @mguetta1. Just want to make sure I have this right. Are you referring to a placeholder for the future or something else? If it's just a placeholder , I think we could leave the code as is and add the dependencies when they do get generated.
> 
> Hi, not a placeholder but an empty array to make sure that this is the expected behavior instead of skipping the `Dependencies` check

Sure, I could do that in a follow up PR . I'd also like to check if analysis_test.go needs to be updated.

**Comment by nachandr** [comment]
> > > > > Thanks @nachandr for the PR. Could you please add analysis tag and dependencies checks?
> > > > 
> > > > 
> > > > Analysis tags and dependencies are not generated with the selected target.
> > > 
> > > 
> > > Good, please set the expected result to an empty array so if something changes we can catch it - `Dependencies: []api.TechDependency{}` Thanks!
> > > Thanks @mguetta1. Just want to make sure I have this right. Are you referring to a placeholder for the future or something else? If it's just a placeholder , I think we could leave the code as is and add the dependencies when they do get generated.
> > 
> > 
> > Hi, not a placeholder but an empty array to make sure that this is the expected behavior instead of skipping the `Dependencies` check
> 
> Sure, I could do that in a follow up PR . I'd also like to check if analysis_test.go needs to be updated.

https://github.com/konveyor/go-konveyor-tests/pull/221/

**Review by mguetta1 — APPROVED** [review]

**Comment by nachandr** [comment]
Hi @aufi and @pranavgaikwad , 
Dependencies and Tags were not found during the analysis of the Apache Wicket app . Is this expected?

**Review by aufi — APPROVED** [review]
Hi @nachandr, this test case specifies path within the source repository, so looking on the sample app: https://github.com/windup/windup-sample-apps/tree/master/test-files/src_example/org/apache/wicket, no dependencies is (probably) correct, there is no pom.xml in the directory subtree.

Generally, there should be tags at least from language discovery (Java and Maven for this app), but these come from language-discovery task triggered automatically with application creation.
So, they might have appeared in case analysis has finished later than the language discovery. This is not common on most applications, but here, the application looks to be just a bunch of `.java` files, so analysis might be pretty fast. This could explain sometimes generated tags.

I tested the analysis locally, except the discovery tags, results are the same and since this is analysis tests more that discovery tests, this PR with empty Dependencies and Tags is LGTM.

**Comment by nachandr** [comment]
> Hi @nachandr, this test case specifies path within the source repository, so looking on the sample app: https://github.com/windup/windup-sample-apps/tree/master/test-files/src_example/org/apache/wicket, no dependencies is (probably) correct, there is no pom.xml in the directory subtree.
> 
> Generally, there should be tags at least from language discovery (Java and Maven for this app), but these come from language-discovery task triggered automatically with application creation. So, they might have appeared in case analysis has finished later than the language discovery. This is not common on most applications, but here, the application looks to be just a bunch of `.java` files, so analysis might be pretty fast. This could explain sometimes generated tags.
> 
> I tested the analysis locally, except the discovery tags, results are the same and since this is analysis tests more that discovery tests, this PR with empty Dependencies and Tags is LGTM.

@aufi , Thank you !
